torch/package/package_exporter.py

```diff
@@ -429,6 +549,43 @@ def register_mock_hook(
         self._mock_hooks[handle.id] = hook
         return handle
 
+    def register_intern_hook(
+        self, hook: Callable[["PackageExporter", str], None]
```

Comment:
Perhaps a type alias would be useful for the hook type? It's repeated in three places now.